src/neps/optimizers/bayesian_optimization/acquisition_samplers/freeze_thaw_sampler.py

```diff
@@ -67,22 +89,47 @@ def _sample_new(
             new_configs, index=range(index_from, index_from + len(new_configs))
         )
 
-    def sample(self, acquisition_function=None, n: int = None) -> pd.Series:
+    def sample(
+            self, 
+            acquisition_function=None, 
+            n: int = None, 
+            set_new_sample_fidelity: int | float=None
+        ) -> list():
+        """Samples a new set and returns the total set of observed + new configs."""
         partial_configs = self.observations.get_partial_configs_at_max_seen()
+        new_configs = self._sample_new(
+            index_from=self.observations.next_config_id(), n=n, ignore_fidelity=False
+        )
 
-        new_configs = self._sample_new(index_from=self.observations.next_config_id(), n=n)
-
-        # Set fidelity for observed configs
+        if self.is_tabular:
+            _n = n if n is not None else self.SAMPLES_TO_DRAW
+            _partial_ids = set([conf["id"].value for conf in partial_configs])
+            _all_ids = set(self.pipeline_space.custom_grid_table.index.values)
+            # accounting for unseen configs only
+            _n = max(_n, len(_all_ids - _partial_ids))
```

Comment:
```suggestion
            _n = min(_n, len(_all_ids - _partial_ids))
```
In the case this is `max` next line will throw an error when `_n>len(_all_ids - _partial_ids)`